packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  buildSearchStatus,
+  SearchStatus,
+  SearchStatusState,
+} from '@coveo/headless';
+import {Component, h, Listen, State, Element} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {Button} from '../../../common/button';
+import {Hidden} from '../../../common/hidden';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+import ArrowBottomIcon from 'coveo-styleguide/resources/icons/svg/arrow-bottom-rounded.svg';
+import {InitPopoverEventPayload} from './popover-event';
+
+/**
+ * @internal
+ * The `atomic-popover` displays a popover menu view of any facet
+ * @part value-button - The button used to display the popover menu
+ * @part value-label - The popover value label
+ * @part arrow-icon - The arrow icon
+ * @part placeholder - The placeholder displayed when the facet isn't ready
+ * @part popover-wrapper - The wrapper that contains the value-button and the slot-wrapper
+ * @part slot-wrapper - The wrapper that contains the facet/slot
+ */
+
+@Component({
+  tag: 'atomic-popover',
+  styleUrl: 'atomic-popover.pcss',
+  shadow: true,
+})
+export class AtomicPopover implements InitializableComponent {
+  @InitializeBindings()
+  public bindings!: Bindings;
+  public searchStatus!: SearchStatus;
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @State()
+  public error!: Error;
+
+  @State() public isMenuVisible = false;
+  @State() public facetId?: string;
+  @State() public facetLabel?: string = 'no-label';
+  @Element() host!: HTMLElement;
+  private facetElement?: HTMLElement;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+  }
+
+  popoverOpened() {
+    if (!this.isMenuVisible) {
+      const popoverOpened = new CustomEvent('popoverOpened', {
```

Comment:
After all, I think that this event should be called something like `clearModals`, since I can imagine actions unrelated to opening a popover wanting to close the popover.

Reply:
I would also make the type of the event more explicit by adding a generic type to represent details and naming the facetId prop.

E.g.:
```ts
interface ClearModalsEventHandler {
  sourcePopoverFacetId?: string;
}
...
const popoverOpened = new CustomEvent<ClearModalsEventHandler>(
  'popoverOpened',
  {
    details: { sourcePopoverFacetId: this.facetId },
  }
);
```

Reply:
I'm not sure I would go with the term modal. I would use dropdown instead.